### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by Dorcas126**
In this project, I start building a school library app. In this initial step, I  implement the classes to represent students and teachers.

# Project Description

- Create a class Person with the following:
    - Instance vars: @id, @name, and @age.
    - Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
    - Getters for @id, @name, and @age.
    - Setters for @name and @age.
    - Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
    - Public method can_use_services? that returns true if a person is of age or if they have permission from parents.
- Create class Student with the following:
     - Inherits from Person.
     - Constructor extends parent's constructor by adding @classroom and a parameter for it.
     - Method play_hooky that returns "¯\(ツ)/¯".
- Create a class Teacher with the following:
     - Inherits from Person.
     - Constructor extends parent's constructor by adding @specialization and a parameter for it.
     - Override can_use_services? so it always returns true.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Person
+  def initialize(id, age, name: 'unknown', parent_permission: true)
+    @id = id
+    @name = name
```
Comment (joel-kalema) [inline person.rb]: - [ ] I like the way you preceding to create your classes 🥇, however, let's generate the `id` Randomly and remove it from `initialize`. It'll be something like this:

```ruby
class Person
 
  attr_accessor :name, :age 
  attr_reader :id 

  def initialize(age, name = "Unknown", parent_permission = true)
    @id = Random.rand(1..1000)
    @age = age
    @name = name
    @parent_permission = parent_permission
  end
  .
  .
  .
end
```

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Student < Person
+  attr_reader :classroom
+
+  def initialize(_id, age, classroom, parent_permission: true, name: 'Unknown')
```
Comment (joel-kalema) [inline student.rb]: - [ ] Then remove `attr_reader :classroom` and id from `initialize`, It'll be something like this:
```ruby
class Student < Person 
  def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end
  
  def play_hooky
    "¯\(ツ)/¯"
  end
end
```
Kindly fix it 💯

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Teacher < Person
+  attr_reader :specialization
+
+  def initialize(_id, age, specialization, parent_permission: true, name: 'Unknown')
```
Comment (joel-kalema) [inline teacher.rb]: - [ ] Also remove `attr_reader :specialization` and `id` from initialize 👍 as it'll be generated automatically, It'll be something like this:

```ruby
class Teacher < Person
 
  def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
  
  def can_use_services?
    true
  end
end
```

Kindly fix it 👍

**Review by joel-kalema — APPROVED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @Dorcas126 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

0bd223a2b156e0eaf61a713e84c3d3b7



## To highlight :champagne: 

- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me @joel-kalema  when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by joel-kalema — CHANGES_REQUESTED** [review]
Keep doing good job 👍

**Comment by Dorcas126** [comment]
> Keep doing good job 👍

did I make the changes required?

**Comment by joel-kalema** [comment]
> > Keep doing good job 👍
> 
> did I make the changes required?

Hi @Dorcas126 
Kindly ask for a second review 👍

**Comment by Dorcas126** [comment]
> > > Keep doing good job 👍
> > 
> > 
> > did I make the changes required?
> 
> Hi @Dorcas126 Kindly ask for a second review 👍

Hi @joel-kalema 
the issue is that you approved, yesterday you approved and same time you requested for changes.

When I wanted to request for second review I found that you approved it.

**Comment by enniopo** [comment]
hi @Dorcas126 - I manually reverted the status of your activity. You can now request another code review request. 

Apologies for the inconvenience.

**Comment by Dorcas126** [comment]
> hi @Dorcas126 - I manually reverted the status of your activity. You can now request another code review request.
> 
> Apologies for the inconvenience.

it's okay , let me resubmit

**Comment by joel-kalema** [comment]
> > > > Keep doing good job 👍
> > > 
> > > 
> > > did I make the changes required?
> > 
> > 
> > Hi @Dorcas126 Kindly ask for a second review 👍
> 
> Hi @joel-kalema the issue is that you approved, yesterday you approved and same time you requested for changes.
> 
> When I wanted to request for second review I found that you approved it.

Please, do not manage it, however, ask for a second review 👍

**Pull request merged**